src/scripts/Controller.js

```diff
@@ -518,7 +521,30 @@ export default class SitemapController {
 			},
 		});
 	}
+	showImportSitemapFile() {
+		function readBlob() {
+			let files = document.getElementById('sitemapFiles').files;
+			if (!files.length) {
+				alert('Please select a file!');
+				return;
+			}
 
+			let file = files[0];
+			let reader = new FileReader();
+			// If we use onloadend, we need to check the readyState.
+			reader.onloadend = function(evt) {
```

Comment:
Check if you can use `reader.onload` here, so you won't have to check for `readyState === FileReader.DONE`.
Also there is new `Blob.text()` API, take a look, it may be more simple.